inception-project/inception
# #1021 - PDF editor performance improvements

**What's in the PR**
* add Aho-Corasick algorithm for span annotation alignment
* adapt tests to work with new methods
* PdfExtractFile and DocumentModel now computed only once a PDF file is opened in PDF editor

**How to test manually**
* open PDF document in PDF editor
* check if annotations render correctly
* check if they are selectable
* check if new annotations can be created

**Automatic testing**
* [X] PR includes unit tests (modifies existing tests to use with new methods)

**Documentation**
